code/modules/station_goals/bluespace_tap.dm

```diff
@@ -428,9 +497,13 @@
 	spawn_time = 30 SECONDS
 	max_mobs = 5		//Dont' want them overrunning the station
 	max_integrity = 250
+	var/obj/machinery/power/bluespace_tap/linked_source_object //this is the BSH that caused it
 
 /obj/structure/spawner/nether/bluespace_tap/deconstruct(disassembled)
 	new /obj/item/stack/ore/bluespace_crystal(loc)	//have a reward
+	if(linked_source_object)
+		linked_source_object.active_nether_portals -= src
+		linked_source_object.update_icon(UPDATE_ICON)
```

Comment:
put this in `Destroy()` instead, if portals ever get deleted via a non `deconstruct()` method then they will not be properly garbage collected